scipy/spatial/ckdtree.pyx

```diff
@@ -923,11 +923,11 @@ cdef class cKDTree:
             result = np.empty(retshape, dtype=object)
             vout = result.reshape(-1)
 
-        vxx = np.reshape(x, (-1, x.shape[-1]))
-        vrr = np.reshape(r, (-1))
+        vxx = np.ascontiguousarray(x).reshape(-1, x.shape[-1])
+        vrr = np.ascontiguousarray(r).reshape(-1)
```

Comment:
Isn't `r` guarunteed to be contiguous by `order='C'` on line 909?

Also, `np.ascontiguousarray` could replace the call to `np.asarray` on line 903.

Reply:
I am only calling np.ascontiguousarray right before casting the array to a memory view object that requires contiguous memory.  So I am leaning towards dropping order='C' to keep this convention. What to do say?

Reply:
What if `vxx` and `vrr` were initialized on the next line from the array initializations?
```python
x = np.ascontiguousarray(x, dtype=np.float64)
vxx = x.reshape(-1, x.shape[-1])
```
That would still be consistent with e.g. `vout` while not duplicating the `asarray` call.

Reply:
Sure I did that; but had to add a few `const` to get it to work. Also lifted out xshape, which makes the code a bit cleaner I think by avoiding some used-only-once variables.